eng/pipelines/templates/steps/resolve-package-targeting.yml

```
@@ -0,0 +1,36 @@
+parameters:
+  - name: BuildTargetingString
+    type: string
+    default: 'azure-*'
+  - name: PackagePropertiesFolder
+    type: string
+    default: ''
+
+steps:
+  # whether we are running within a PR build or not, we need to walk the package properties and save them to
+  # variable $(TargetingString) so that every other check and accept it as an argument.
+  - pwsh: |
+      $setting = "${{ parameters.BuildTargetingString }}"
+
+      # if the variable is not set, it'll just come back as the variable name. otherwise it's set.
+      if ('$(BuildTargetingString)' -ne ('$' + '(BuildTargetingString)')) {
+          Write-Host "The variable named BuildTargetingString is set to $(BuildTargetingString)"
+          $setting = "$(BuildTargetingString)"
+      }
+      else {
+        $packageProperties = Get-ChildItem -Recurse -Force "${{ parameters.PackagePropertiesFolder }}" `
+          | Where-Object { $_.Extension -eq '.json' } `
+          | Foreach-Object { $_.Name } `
+          | ForEach-Object { $_.Replace(".json", "") }
```

```suggestion
        $packageProperties = Get-ChildItem -Recurse -Force "${{ parameters.PackagePropertiesFolder }}/*.json" `
          | ForEach-Object { $_.Name.Replace(".json", "") }
```

follow-up: potential clean-up.